pkg/tools/sandbox/registry_client.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sandbox
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice/payload"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/common/interpose"
+	"github.com/networkservicemesh/sdk/pkg/registry/common/null"
+	"github.com/networkservicemesh/sdk/pkg/registry/common/refresh"
+	"github.com/networkservicemesh/sdk/pkg/registry/core/chain"
+)
+
+type registryClient struct {
+	nsClient  registry.NetworkServiceRegistryClient
+	nseClient registry.NetworkServiceEndpointRegistryClient
+}
+
+// NewRegistryClient returns NSE registry client to register NSE to registry
+func NewRegistryClient(ctx context.Context, cc grpc.ClientConnInterface, isForwarder bool) registry.NetworkServiceEndpointRegistryClient {
```

Comment:
Can we just create chains for registry clients?

Reply:
Yep, done.